tests/components/advantage_air/test_climate.py

```diff
@@ -169,6 +171,19 @@ async def test_climate_async_setup_entry(hass, aioclient_mock):
     assert aioclient_mock.mock_calls[-1][0] == "GET"
     assert aioclient_mock.mock_calls[-1][1].path == "/getSystemData"
 
+    # Test set_myair service
+    await hass.services.async_call(
+        ADVANTAGE_AIR_DOMAIN,
+        ADVANTAGE_AIR_SERVICE_SET_MYZONE,
+        {ATTR_ENTITY_ID: [entity_id]},
+        blocking=True,
+    )
+    assert len(aioclient_mock.mock_calls) == 17
+    assert aioclient_mock.mock_calls[-2][0] == "GET"
+    assert aioclient_mock.mock_calls[-2][1].path == "/setAircon"
+    assert aioclient_mock.mock_calls[-1][0] == "GET"
+    assert aioclient_mock.mock_calls[-1][1].path == "/getSystemData"
```

Comment:
I would have expected here to be an assert that verifies if the zone has actually been successfully set? Am I missing something?

Reply:
Agree, I would expect validation of the zone number here.
@Bre77 Could you take another look?

Reply:
The integration is currently not "optimistic", changes to values are only done by reading the response from the API, which I cannot do with a mock request.

So can I clarify what the ask is here? Is it to make the service optimistic and change the value in hass before it's accepted by the device?

Reply:
You are checking the mock calls, you should also check if the zone ended up in that mock.

Reply:
@frenck @spacegaier this was definitely an oversight on my behalf, I have added the assertion.